cpp/src/components/connectivity.cu

```diff
@@ -131,40 +146,85 @@ gdf_connected_components_impl(gdf_graph *graph,
     }
   else
     {
-      //dump error message and return unsupported, for now:
+      
+      //device memory requirements: 2n^2 + 2n x sizeof(IndexT) + 1 (for flag)
+      //( n = |V|)
       //
-      std::cerr<<"ERROR: Feature not supported, yet;"
-               <<" at: " << __FILE__ << ":" << __LINE__ << std::endl;
+      size_t n2 = 2*nrows;
+      n2 = n2*(nrows*sizeof(ByteT) + sizeof(IndexT)) + 1;
+
+      int device;
+      cudaDeviceProp prop;
+      
+      cudaGetDevice(&device);
+      cudaGetDeviceProperties(&prop, device);
+
+      if( n2 > prop.totalGlobalMem )
+        {
```

Comment:
What's the purpose of this check (we need to check memory allocation results anyway)?

"n2 > prop.totalGlobalMem" does not guarantee that we have enough memory to finish this computation (e.g. most of the totalGlobalMem can be already allocated to other GPU dataframes unrelated to this function).

Reply:
The purpose of the check is to ensure _minimal_ memory requirements. Since this implementation of SCC is very memory consuming (O(|V|^2), the algorithm may fail. Without this check it would fail silently without apparent reason. This was a requirement that was specifically agreed upon during the numerous talks about GEMM SCC implementation and specifically required by @BradReesWork .

Reply:
Does this mean this can still silently fail if memory allocation fails?